File: simchodi/로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    answer = []
    rank_list = {2:5, 3:4, 4:3, 5:2}
    zero_count = lottos.count(0)
    same_count = 0
    for win in win_nums:
        if win in lottos:
            same_count += 1
    zero_count += same_count
    
    if zero_count >= 6:
        answer.append(1)        
    elif zero_count <= 1:
        answer.append(6)        
    else:
        for rank, val in rank_list.items():
            if zero_count == val:
                answer.append(rank)
    
    if same_count >= 6:
        answer.append(1)        
    elif same_count <= 1:
        answer.append(6)        
    else:
        for rank, val in rank_list.items():
            if same_count == val:
                answer.append(rank)
                
    return answer

File: thyoondev/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  const rank = { 6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6 };
  let total = 0;
  let zero = 0;
  lottos.map((n) => (win_nums.includes(n) ? total++ : n === 0 ? zero++ : n));
  return [rank[total + zero], rank[total]];
}
